canvas_sdk/utils/http.py

```diff
@@ -0,0 +1,55 @@
+import time
+from functools import wraps
+from typing import Any, Callable, TypeVar
+
+import requests
+import statsd
+
+F = TypeVar("F", bound=Callable)
+
+
+class Http:
+    """A helper class for completing HTTP calls with metrics tracking."""
+
+    def __init__(self) -> None:
+        self.session = requests.Session()
+        self.statsd_client = statsd.StatsClient()
+
+    @staticmethod
+    def measure_time(fn: F | None = None) -> Callable[[F], F] | F:
+        """A decorator to store timing of HTTP calls."""
+
+        def _decorator(fn: F) -> F:
+            @wraps(fn)
+            def wrapper(self: "Http", *args: Any, **kwargs: Any) -> Any:
+                print(fn.__name__)
+                start_time = time.time()
+                result = fn(self, *args, **kwargs)
+                end_time = time.time()
+                timing = int((end_time - start_time) * 1000)
+                self.statsd_client.timing(f"http_{fn.__name__}", timing)
+                return result
+
+            return wrapper
+
+        return _decorator(fn) if fn else _decorator
+
+    @measure_time
+    def get(self, url: str, headers: dict = {}) -> requests.Response:
+        """Sends a GET request."""
+        return self.session.get(url, headers=headers)
+
+    @measure_time
+    def post(self, url: str, json: dict | None = None, headers: dict = {}) -> requests.Response:
```

Comment:
Do we need to allow post, put, and patch to send non-JSON request bodies?

Reply:
i think we want to keep it simple for now, and if there is a need to add in more other uses like those (files, etc), we'll add them. it'll be simple enough since we're just copying the same api as the requests library and passing in.

Reply:
OK. The specific non-JSON use case I was thinking about was for request bodies that are  `application/x-www-form-urlencoded` -- like requesting an OAuth token.

Reply:
that's a good one! i'll add it